crates/router/src/core/payments/retry.rs

```diff
@@ -358,6 +358,16 @@ where
                 .change_context(errors::ApiErrorResponse::InternalServerError)
                 .attach_printable("Could not parse the connector response")?;
 
+            let additional_payment_method_data =
+                payments::helpers::update_additional_payment_data_with_connector_response_pm_data(
+                    payment_data.payment_attempt.payment_method_data.clone(),
+                    router_data
+                        .connector_response
+                        .and_then(|connector_response| {
+                            connector_response.additional_payment_method_data
+                        }),
+                )?;
+
```

Comment:
This can be moved out of match statement and `additional_payment_method_data` can be used in match arms

Reply:
Yes, it can be moved. I had thought about moving it to a common place. Not sure why I decided back to do it in both the match arms. Nevertheless, will move it outside this match arm